cloudevents-sdk-rdkafka/src/kafka_consumer_record.rs

```diff
@@ -0,0 +1,281 @@
+use crate::headers;
+use cloudevents::event::SpecVersion;
+use cloudevents::message::{
+    BinaryDeserializer, BinarySerializer, Encoding, MessageAttributeValue, MessageDeserializer,
+    Result, StructuredDeserializer, StructuredSerializer,
+};
+use cloudevents::{message, Event};
+use rdkafka::message::{BorrowedMessage, Headers, Message, OwnedMessage};
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::str;
+
+pub struct ConsumerRecordDeserializer {
+    pub(crate) headers: HashMap<String, Vec<u8>>,
+    pub(crate) payload: Vec<u8>,
+}
+
+impl ConsumerRecordDeserializer {
+    pub fn owned_new(message: OwnedMessage) -> ConsumerRecordDeserializer {
+        let mut resp_des = ConsumerRecordDeserializer {
+            headers: HashMap::new(),
+            payload: Vec::new(),
+        };
+        let headers = message.headers().unwrap();
+        for i in 0..headers.count() {
+            let header = headers.get(i).unwrap();
+            resp_des
+                .headers
+                .insert(header.0.to_string(), Vec::from(header.1));
+        }
+
+        match message.payload() {
+            Some(s) => resp_des.payload = Vec::from(s),
+            None => resp_des.payload = resp_des.payload,
```

Comment:
Do you really need this match arm?

Reply:
this is one of those "ill deal with later" things i was talking about. Has been sorted